docs/proposals/storage_secrets.md

```diff
@@ -0,0 +1,94 @@
+<!-- BEGIN MUNGE: UNVERSIONED_WARNING -->
+
+<!-- BEGIN STRIP_FOR_RELEASE -->
+
+<img src="http://kubernetes.io/img/warning.png" alt="WARNING"
+     width="25" height="25">
+<img src="http://kubernetes.io/img/warning.png" alt="WARNING"
+     width="25" height="25">
+<img src="http://kubernetes.io/img/warning.png" alt="WARNING"
+     width="25" height="25">
+<img src="http://kubernetes.io/img/warning.png" alt="WARNING"
+     width="25" height="25">
+<img src="http://kubernetes.io/img/warning.png" alt="WARNING"
+     width="25" height="25">
+
+<h2>PLEASE NOTE: This document applies to the HEAD of the source tree</h2>
+
+If you are using a released version of Kubernetes, you should
+refer to the docs that go with that version.
+
+Documentation for other releases can be found at
+[releases.k8s.io](http://releases.k8s.io).
+</strong>
+--
+
+<!-- END STRIP_FOR_RELEASE -->
+
+<!-- END MUNGE: UNVERSIONED_WARNING -->
+
+## Abstract
+
+In Kubernetes, the API object 'secrets' provide a means to store confidential credentials in a volume to be accessed by 
+the Kuberenetes Master as well as other external resources for the pod, such as database login information. 
+
+Creating a secret volume for the storage layer has use cases outside the norm of traditional access.  Today, the
+access to the secret is typically contrainted to a namepsace and application within that namespace requiring the information.
+Secrets are created for a purpose that can be isolated by one function alone. Storage security presents the obstacle that secrets 
+with actual storage assets may not have a specific namespace when the volumes are created. Therefore, facilitiating the need for an
+alternate approach.
+
+## Motivation
+
+Today, ACLs are not well defined in storage. An admin cannot create precise contraints for a user or group based on the actual storage
+assets and have it flow through the system. In addition, administrators whom pre-create volumes to expedite the process or to limit users, don't know which user or group may want to request that resource. Since secrets today are namespace constrainted, we cannot attach the
+secret globally. Nor, would that solution necessarily be desireable considering the possibility of exposure.
+
```

Comment:
How secure are secrets today? I know they are used to hold passwords in some of the examples, but are they viewed as being sufficiently secure to contain storage access credentials?